conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -224,23 +224,36 @@ def update_autoreconf_args(self, updated_flags):
 
     # FIXME: Remove all these update_xxxx whenever xxxx_args are dicts or new ones replace them
     def _update_flags(self, attr_name, updated_flags):
-        _new_flags = []
-        self_args = getattr(self, attr_name)
-        for index, flag in enumerate(self_args):
-            flag_name = flag.split("=")[0]
-            if flag_name in updated_flags:
-                new_flag_value = updated_flags[flag_name]
-                # if {"build": None} is passed, then "--build=xxxx" will be pruned
-                if new_flag_value is None:
-                    continue
-                elif not new_flag_value:
-                    _new_flags.append(flag_name)
+
+        def _list_to_dict(flags):
+            ret = {}
+            for flag in flags:
+                # Only splitting if "=" is there
+                option = flag.split("=")
+                if len(option) == 2:
+                    ret[option[0]] = option[1]
                 else:
-                    _new_flags.append(f"{flag_name}={new_flag_value}")
-            else:
-                _new_flags.append(flag)
+                    ret[option[0]] = ""
+            return ret
+
+        def _dict_to_list(flags):
+            ret = []
+            for k, v in flags.items():
+                # None values are pruned from the global result
+                if v is None:
+                    continue
+                # If value, it's assumed the flag will be flag=value, else keeping flag only
+                # For instance: {"--opt1": "whatever", "-abc": ""} --> ["--opt1=whatever", "-abc"]
+                ret.append(f"{k}={v}" if v else k)
+            return ret
```

Comment:
```suggestion
            return [f"{k}={v}" if v else k for k, v in flags.items() if v is not None]
```
And as it is only used once, just do not add a function, but have it inline (the less jumps, the easier to read)

Reply:
Sometimes, I prefer to read several lines instead a quite-compressed one, but, yes, in this case, I believe it's easy to read.

Reply:
Depends on the complexity, the semantics, the number of re-uses...
Typical things I check:
- If it is a oneliner or two liner
- If is using typical list comprehension patterns
- If the intention can be summarized in one line comment.
In this case, something like this reads better than having to jump to a function and read many lines:
```python
... # flags computed here
# Flatten flags dict to a list of ``k=v`` or ``k`` removing None values
my_list = [f"{k}={v}" if v else k for k, v in flags.items() if v is not None]
# use my list
```